pkg/delete/delete.go

```diff
@@ -46,37 +46,56 @@ type DeleteEntries = map[configurationType][]pointer.DeletePointer
 
 // Configs removes all given entriesToDelete from the Dynatrace environment the given client connects to
 func Configs(ctx context.Context, clients ClientSet, apis api.APIs, automationResources map[string]config.AutomationResource, entriesToDelete DeleteEntries) error {
-	deleteErrors := 0
-	for entryType, entries := range entriesToDelete {
-		if entryType == api.DashboardShareSettings {
-			log.Warn("Classic config of type %s cannot be deleted. Note, that they can be removed by deleting the associated dashboard.", api.DashboardShareSettings)
+	var deleteErrors int
+
+	// Delete automation resources (in the specified order
+	automationTypeOrder := []string{"workflow", "scheduling-rule", "business-calendar"}
+	for _, key := range automationTypeOrder {
+		entries := entriesToDelete[key]
+		if reflect.ValueOf(clients.Automation).IsNil() {
+			log.WithCtxFields(ctx).WithFields(field.Type(key)).Warn("Skipped deletion of %d Automation configuration(s) of type %q as API client was unavailable.", len(entries), key)
+			delete(entriesToDelete, key)
 			continue
 		}
+		err := automation.Delete(ctx, clients.Automation, automationResources[key], entries)
+		if err != nil {
+			log.WithFields(field.Error(err)).Error("Error during deletion: %v", err)
+			deleteErrors += 1
+		}
+		delete(entriesToDelete, key)
+	}
 
-		var err error
+	// Delete bucket resources
+	if reflect.ValueOf(clients.Buckets).IsNil() {
```

Comment:
nit: other than the reflection, and the fact that we can access it directly, is there a reason to handle bucket deletion separate to configs or settings? We could also just delete the bucket entries if no client is defined and leave the delete call in the main loop over delete entries